packages/flutter/lib/src/material/icon.dart

```diff
@@ -65,13 +83,13 @@ class Icon extends StatelessComponent {
       category = parts[0];
       subtype = parts[1];
     }
-    // TODO(eseidel): This clearly isn't correct.  Not sure what would be.
-    // Should we use the ios images on ios?
-    String density = 'drawable-xxhdpi';
+    // TODO(kgiesing): Should selector logic be moved down a level so that
+    // any generic asset can be resolved using consistent contextual logic?
```

Comment:
Right, that makes sense. In particular the mapping of DPI to selector string seems like it should move down.

If we have both generic and specific mapping logic, there's the question of whether the material logic supplements it or replaces it. If it's going to supplement it, then the density selector part might want to move out of the AssetImage name.